bin/utest/pipe.c

```diff
@@ -0,0 +1,91 @@
+#include <assert.h>
+#include <unistd.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <errno.h>
+
+#include <signal.h>
+#include <string.h>
+
+#include <sys/types.h>
+#include <sys/wait.h>
+
+#include "utest.h"
+#include "util.h"
+
+static sig_atomic_t parent_signal_delivered = 0;
+static sig_atomic_t child_signal_delivered = 0;
+
+void parent_sigpipe_handler(int signo) {
+  parent_signal_delivered = 1;
+  return;
+}
+
+void child_sigpipe_handler(int signo) {
+  child_signal_delivered = 1;
+  return;
+}
+
+int test_pipe_parent_signaled(void) {
+  int pipe_fd[2];
+
+  assert(pipe2(pipe_fd, 0) == 0);
+
+  pid_t child_pid = fork();
+  assert(child_pid > 0);
+
+  if (child_pid == 0) { /* child */
+    close(pipe_fd[1]);  /* closing write end of pipe */
+    close(pipe_fd[0]);  /* closing read end of pipe */
+    exit(EXIT_SUCCESS);
+  }
+
+  /* parent */
+  close(pipe_fd[0]); /* closing read end of pipe */
+
+  // Sync with end of child execution
+  wait_for_child_exit(child_pid, EXIT_SUCCESS);
+
+  // This is supposed to trigger SIGPIPE and return EPIPE
+  assert(write(pipe_fd[1], "hello world\n", 12) == EPIPE);
```

Comment:
Does `write` return -1 or `EPIPE`? Manual says that it will return -1 and set `errno` to appropriate value.

Reply:
You're right, I forgot to assure myself with manual